### ISSUE TITLE
[naming-convention] modifier global - Allow selecting only global variables

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#2318]
Similar to #2239.

I'd like to be able to do something similar to [this](https://github.com/ajafff/tslint-consistent-codestyle/blob/master/docs/naming-convention.md#examples), i.e.:

```js
// require all global constants to be camelCase or UPPER_CASE
// all other variables and functions still need to be camelCase
{"selector": "variable", "modifiers": ["global", "const"], "format": ["camelCase", "UPPER_CASE"]}
```

**Comment by InsOpDe** [comment]
Its funny because the description of the `naming-convention` rule states

> Additionally, a well-designed style guide can help communicate intent, such as by enforcing all private properties begin with an _, and all global-level constants are written in UPPER_CASE.

and yet its not able to enforce this rule

**Comment by bradzacher** [comment]
Happy to accept a PR if you want it.

**Comment by phaux** [comment]
Maybe the name `"topLevel"` or something similar would be more appropriate for the modifier? When I hear "global" I think something attached to the window object. The convention I personally use is to name every top level or exported const with upper snake case, but use camel case for the ones in blocks, functions, classes, etc.

**Comment by phaux** [comment]
Unless you actually meant "global" as in attached to the window, but then it can't be a "const" (at least in modules)

**Comment by bradzacher** [comment]
This should be relatively easy to do, if anyone wants to try.

- define a new modifier
- when you encounter a variable definition - get the scope (`context.getScope()`)
- check `scope.type` is either `global` or `module`

**Comment by infacto** [comment]
Just to visualize a use case: 

```TS
// Global variable allowed to be UPPER_CASE.
const MY_VALUE = 42;

class MyClass {
  myMethod() {
    // This local const (variable) is NOT allowed.
    const ANOTHER_VALUE = 27;

    // This is allowed.
    const anotherValue = 13;
  }
}
```
Since `const` has different meaning in TypeScript then in other languages. If a variable (`let`) is never reassigned `const` should be used. (rule `prefer-const`, btw. which should be warnings instead of errors by default. anyway...) I only want `strictCamelCase` for that variables. But allow `UPPER_CASE` for "real" constants and e.g. `static readonly` (which already work) The current workaround is either disallow it or allow both cases and mention it in the code reviews.
